### ISSUE TITLE
508-defect-2 [COGNITION]: "Get directions" link should be more descriptive and inform users if it will open in a new tab

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#25006]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As any user (but especially AT users), I want to know if a link will open a new tab and where it will link me to.

## Details

"Get directions" does not provide key information on _where_ directions will be procured from and that it opens a new tab. The latter can be particularly frustrating if unanticipated by assistive tech users.

## Acceptance Criteria

- [ ] Rewrite "get directions" link on the main facility page to "Directions on Google Maps"
- [ ] Rewrite all "get directions" links on the locations page to be more specific "Directions to XYZ Center on Google Maps"
- [ ] Clicking on "Directions to Google Maps" link does not open a new tab

## Screenshots or Trace Logs

<img width="754" alt="Screen Shot 2021-05-20 at 10 34 17 AM" src="https://user-images.githubusercontent.com/14154792/118998881-cd255000-b957-11eb-9df8-d6b408ca2160.png">
<img width="621" alt="Screen Shot 2021-05-20 at 10 35 51 AM" src="https://user-images.githubusercontent.com/14154792/118998884-cdbde680-b957-11eb-8467-fc99f29e91d6.png">

https://user-images.githubusercontent.com/14154792/119009300-fbf3f400-b960-11eb-802e-41eae0da3654.mov

**Comment by mmiddaugh** [comment]
Josh's explanation from [Slack thread](https://dsva.slack.com/archives/C0FQSS30V/p1623076761008700?thread_ts=1622846087.497600&cid=C0FQSS30V)

**Directions on google maps (opens in new tab) should be the experience for sighted and screen reader users.**

- Sighted keyboard users need a heads up as to how this link will behave if it opens in a new tab, so opens in new tab shouldn’t be sr-only
- Everyone would benefit from knowing this opens in google maps so they have a heads up as to where it will take them and will set expectations appropriately
- That being said, (Opens in new tab) is only necessary if that’s the behavior of the link. Unless there’s a significant design reason to make this open in a new tab, a more accessible pattern would be to not make it open in a new tab (which respects the user’s agency and choice). If we were to set the behavior to be a standard link instead of one that opens in a new tab, we could remove it from the string.

**Name of the vet center is more screen reader specific for folks who navigate using the links list.** 

- The links list is a literally a list of links they can hear to navigate through the page. If every link says “Directions on google maps (opens in new tab)“, it won’t be very useful. That being said, it’s not as important to sighted users as they can visibly see where the directions are located under.

**To conclude, three solutions exist here:**

1. Write the visible and screen reader link as “Directions to [Vet Center] on google maps (opens in new tab)” and not worry about using an aria-label or sr-only text
2. Write the visible link as “Directions on google maps (opens in new tab)” and use aria-label or add sr-only text that include the Vet Center’s name
3. Revert the behavior to not open in a new tab, and remove the “opens in new tab” string from the link

**Comment by lildziner** [comment]
On this VC locations page, I could still use my back button and not ‘lose my place’ so opening a new tab might not be as necessary. I'd be ok with it not opening a new tab here.

Opening in a new tab is more relevant on Facility Locator where you’ve gone through the work of setting your search parameters and if you’re comparing locations and directions. My assumption is that opening new tabs is more forgiving of a person in exploration mode. It may also be helpful to open in another tab since you’re leaving VA.gov. It’s also how many other facility locators on the web work (new tab).

Also, I think it's overkill to visibly state “opens in a new tab” if we’re already stating “Directions on Google Maps”. I would prefer to start with the visible change to “Directions on Google Maps” and revisit the additional text if we learn Veterans are having trouble with this behavior.

**Comment by joshkimux** [comment]
Great context here! If @lildziner is comfortable with it, I'd also recommend not opening in a new tab and removing it from the visible string. That would meet WCAG guidance for [G200: Opening new windows and tabs from a link only when necessary](https://www.w3.org/TR/WCAG20-TECHS/G200.html) as this is a case where it is not necessary.

**Comment by lildziner** [comment]
I'm comfortable removing the open a new tab behavior here. Thanks, @joshkimux

**Comment by mpelzsherman** [comment]
My understanding is that the new AC are as follows:

For all of Vet Center facility directions links:

- Remove the facility name
- Add an aria-label attribute that includes the facility name
- Remove the (opens in new tab) message and behavior

I will proceed with implementing these new requirements until I hear otherwise.

**Comment by joshkimux** [comment]
Correct! Thank you for the help @mpelzsherman 
If possible, make sure that the `aria-label` includes the rest of the string too e.g. `aria-label="Directions to [Vet Center] on Google Maps"`
